proof_test.go

```diff
@@ -10,22 +10,22 @@ import (
 
 func computeLeafHash(ap *AuthenticationPath) (leafHash []byte) {
 	leaf := ap.Leaf()
-	if !leaf.IsEmpty() {
+	if !ap.Leaf().IsEmpty() {
```

Comment:
Why do you use ``ap.Leaf()`` here and below if you have ``leaf := ap.Leaf()`` above?